### PR TITLE
Oura: Handle cases where data type is not available

### PR DESCRIPTION
**Pull request by zwkohn**
## Overview

https://github.com/CareEvolution/deploy-myfhr/issues/4929#issuecomment-2776563684
This handles the cases for combined steps/sleep/heartrate where Oura is enabled on the project but the necessary data type is not available

## Security

> Consider potential security impacts and complete the following checklist. 
> **REMINDER:** All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Using the existing MDH calls to determine if data types are available.

## Testing

> Consider whether the changes might have device-specific behaviors (screen padding, new APIs, etc.) and check one of the following boxes:

- [x] This change can be adequately tested using the MDH Storybook.
- [ ] This change requires additional testing in the MDH iOS/Android/Web apps. (Create a pre-release tag/build and test in a ViewBuilder PR.)

This can be tested w/ Storybook utilizing an access token and using the Steps Live Bar component. The Project for this participant should have Oura Enabled in ProjectSettings, but not have "daily-activity" enabled in RealTimeDeviceDataConfiguration for this project. The component should show "No Data" and no errors should occur in browser console.

### Documentation

> Consider whether there are any documentation impacts and check one of the following boxes:

- [ ] I have added relevant Storybook updates to this PR.
- [ ] If this feature requires a developer doc update, I have tagged `@CareEvolution/api-docs`.
- [x] This change does not impact documentation or Storybook.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

### CODE REVIEW

**Review by greinard — COMMENTED** [review]
This looks pretty good to me, but I wanted to surface a few things that came up while looking through this.

- This will likely create some non-trivial merges to work through for https://github.com/CareEvolution/MyDataHelpsUI/pull/429.

- This update fixes the querying performed by the combined providers, but it doesn't fix the "availability check" for the combined providers, which will still error out if it gets far enough down to query for Oura data.  The availability checks are addressed by https://github.com/CareEvolution/MyDataHelpsUI/pull/429, but that needs to be updated to include this new additional type checking for the specific Oura data types.

- I had forgotten that all of the V2 data types come back from the API, enabled or not.  For V1, the list only includes enabled (queryable) data types.  I think it could be error prone for us to remember to always check the `enabled` flag on the V2 types, so I submitted this PR: https://github.com/CareEvolution/MyDataHelps.js/pull/157.  That should allow us to call `MyDataHelps.getDeviceDataV2AllDataTypes(true)`, which will return only the enabled data types.  However, until that is available, we will need to continue checking the enabled flag on each data type as well.

- Since we are having to check the enabled flag for the time being, would you mind adding an additional test case to each of the combined provider tests below to show what happens if Oura is enabled, Oura V2 data type(s) exist, but none are enabled.

File: src/helpers/daily-data-providers/combined-resting-heart-rate.ts [hunk 1]
```diff
@@ -6,14 +6,19 @@ import MyDataHelps from "@careevolution/mydatahelps-js";
 export default function (startDate: Date, endDate: Date) {
 	let providers: Promise<{ [key: string]: number }>[] = [];
 
-	return MyDataHelps.getDataCollectionSettings().then((settings) => {
+	return Promise.all([MyDataHelps.getDataCollectionSettings(), MyDataHelps.getDeviceDataV2AllDataTypes()]).then(([settings, deviceDataV2Types]) => {
```
Comment (bxh) [inline src/helpers/daily-data-providers/combined-resting-heart-rate.ts:9]: I introduced a combined data collection setting helper API that exactly does this in https://github.com/CareEvolution/MyDataHelpsUI/pull/429/files. But it is the server side that should have provided a single endpoint.

Comment (bxh) [inline src/helpers/daily-data-providers/combined-resting-heart-rate.ts:9]: You can use `MyDataHelps.getDeviceDataV2AllDataTypes(true)`, which only returns enabled data types.

File: src/helpers/daily-data-providers/combined-resting-heart-rate.ts [hunk 2]
```diff
@@ -6,14 +6,19 @@ import MyDataHelps from "@careevolution/mydatahelps-js";
...
 		if (settings.fitbitEnabled) {
 			providers.push(fitbitRestingHeartRateDataProvider(startDate, endDate));
 		}
 		if (settings.garminEnabled) {
 			providers.push(garminRestingHeartRateDataProvider(startDate, endDate));
 		}
-		if (settings.ouraEnabled) {
+		if (settings.ouraEnabled && deviceDataV2Types.some(
+			ddt =>
+				ddt.enabled &&
```
Comment (bxh) [inline src/helpers/daily-data-providers/combined-resting-heart-rate.ts:18]: With `MyDataHelps.getDeviceDataV2AllDataTypes(true)`, you can remove this clause.

**Comment by bxh** [comment]
https://github.com/CareEvolution/MyDataHelpsUI/pull/429 introduced a lot of merge conflicts here. If you don't mind, I can merge this PR. for you.

**Comment by bxh** [comment]
All good now. @zwkohn

I switched to use `getCombinedDataCollectionSettings` from https://github.com/CareEvolution/MyDataHelpsUI/pull/429 to updated all the tests.

Feel free to merge after the build turns green.
